mockbroker.go

```diff
@@ -192,8 +215,10 @@ func (b *MockBroker) handleRequests(conn net.Conn, idx int, wg *sync.WaitGroup)
 		}
 
 		b.lock.Lock()
+		originalHandlerUsed := b.origHandler
```

Comment:
It makes sense to have that logic into the handler but the current handler profile do not provide the request size.
Changing the handler function profile to provide such information would help but it will break backward compatibility.
But then we would need the response size as well, I guess it can be computed in the handler before returning it but in case the Write fails the data would be inconsistent.

The `origHandler` field is used for signaling when all the expected requests have been fully processed through a `WaitGroup`.

Maybe having just an optional notifier function called after a request has been handled is the way to go to separate concerns, signal when requests have been processed and avoid braking compatibility:
```
type requestHandledNotifierFunc func(err error, int bytesRead, int bytesWritten)
```